packages/config-loader/src/lib/resolver.ts

```diff
@@ -15,21 +15,45 @@
  */
 
 import { resolve as resolvePath } from 'path';
+import { pathExists } from 'fs-extra';
 
 type ResolveOptions = {
-  // Root path for search for app-config.yaml
-  rootPath: string;
+  // Root paths to search for config files. Config from earlier paths has lower priority.
+  rootPaths: string[];
+  // The environment that we're loading config for, e.g. 'development', 'production'.
+  env: string;
 };
 
 /**
  * Resolves all configuration files that should be loaded in the given environment.
+ *
+ * For each root directory, search for the default app-config.yaml, along with suffixed
+ * NODE_ENV and local variants, e.g. app-config.production.yaml or app-config.development.local.yaml
+ *
+ * The priority order of config loaded through suffixes is `env > local > none`, meaning that
+ * for example app-config.development.yaml has higher priority than `app-config.local.yaml`.
+ *
  */
 export async function resolveStaticConfig(
   options: ResolveOptions,
 ): Promise<string[]> {
-  // TODO: We'll want this to be a bit more elaborate, probably adding configs for
-  //       specific env, and maybe local config for plugins.
-  const configPath = resolvePath(options.rootPath, 'app-config.yaml');
+  const filePaths = [
+    `app-config.yaml`,
+    `app-config.local.yaml`,
+    `app-config.${options.env}.yaml`,
+    `app-config.${options.env}.local.yaml`,
```

Comment:
hm not sure if needed - like, would you use a testing.local for example?

Reply:
Bit just for completeness, but it is needed if you want to override things in an env config, as the env config has higher prio than local. Could ofc consider changing that too though :grin: